frontend/src/components/Field/Attachment/utils/getInvalidAttachments.ts

```diff
@@ -0,0 +1,57 @@
+import JSZip from 'jszip'
+import { flattenDeep, uniq } from 'lodash'
+
+import { AttachmentProps } from '../Attachment'
+
+import { getFileExtension } from './getFileExtension'
+
+/**
+ * Checks whether the given file extension is valid against the list of valid
+ * extensions.
+ *
+ * @param ext the file extension to check
+ * @return `true` if the file extension is invalid, otherwise `false`.
+ */
+export const isInvalidFileExtension = (
+  ext: string,
+  accept: AttachmentProps['accept'],
+): boolean => {
+  // Every extension is valid if accept array is undefined/empty.
+  if (!accept) return false
+  return !accept.includes(ext.toLowerCase())
+}
+
+/**
+ * Dives into a zip file and recursively checks if it contains
+ * any invalid files. A file is deemed invalid if its file extension
+ * is not valid as checked by isInvalidFileExtension.
+ *
+ * @param file the file to check
+ * @return array of invalid file extensions in given zip file
+ */
+export const getInvalidFileExtensionsInZip = (
```

Comment:
Why would memos need to support zips

Reply:
This attachment dropzone is more or less lifted from Forms. I've just left this zip support in for now.